src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,29 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.writeLock().lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+        }
```

Comment:
what about the read operations, should we allow them while we add/modify the data?